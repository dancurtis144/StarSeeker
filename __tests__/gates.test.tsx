import { render } from "@testing-library/react-native";

import GateScreen from "@/app/(tabs)/gates";

describe("GateScreen", () => {
  it("renders initial screen correctly", () => {
    const { getByText } = render(<GateScreen />);
    expect(getByText("Gates")).toBeTruthy();
    expect(getByText("No gates available")).toBeTruthy();
  });

  it("matches snapshot", () => {
    const screen = render(<GateScreen />);
    expect(screen).toMatchSnapshot();
  });
});
