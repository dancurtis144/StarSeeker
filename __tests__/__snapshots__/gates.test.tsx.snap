// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GateScreen matches snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "flexDirection": "row",
          "gap": 8,
          "paddingHorizontal": 32,
          "paddingTop": 80,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "lineHeight": 32,
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Gates
    </Text>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      {
        "gap": 20,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={10}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "flex": 1,
          "padding": 20,
        },
        {
          "backgroundColor": "#fff",
        },
      ]
    }
    viewabilityConfigCallbackPairs={[]}
    windowSize={2}
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          No gates available
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
